client/index.js

```diff
@@ -47,7 +47,7 @@ function log(level, msg) {
 
 // Send messages to the outside, so plugins can consume it.
 function sendMsg(type, data) {
-	if(typeof self !== "undefined" && self.window) {
+	if(typeof self !== "undefined" && typeof WorkerGlobalScope === "undefined") {
```

Comment:
I'd like to amend this slightly:

```js
if(typeof self !== "undefined" && typeof WorkerGlobalScope === "undefined" && !self instanceOf WorkerGlobalScope) {
```

If we find another reason to check to see if the script is executing within a WebWorker, we'll want to move that out to a shared var.